lib/index.js

```diff
@@ -53,12 +53,23 @@ module.exports = {
             if (err) {
               return reject(err);
             }
-            // Set the bucket file url. 
-            // If there is a Custom endpoint for data access set, replace the upload endpoint with the read enpoint URL
-            if(config.cloudflarePublicAccessUrl){
-              file.url = config.cloudflarePublicAccessUrl.replace(/\/$/g,'')+'/'+data.Key;
+
+            // When a file is >5MB, Location is set to "auto" and the bucket name is prepended to the key.
+
+            // Strip bucket name from key if Location is auto
+            const key = data.Location === 'auto' && data.Key.startsWith(`${config.params.Bucket}/`)
+              ? data.Key.replace(`${config.params.Bucket}/`, '')
+              : data.Key;
+
+            // Set the bucket file URL.
+            // If there is a custom endpoint for data access set, replace the upload endpoint with the read enpoint URL.
+            // Otherwise, use location returned from S3 API if it's not "auto"
+            if (config.cloudflarePublicAccessUrl) {
+              file.url = config.cloudflarePublicAccessUrl.replace(/\/$/g, '') + '/' + key;
+            } else if (data.Location !== 'auto') {
+              file.url = data.Location;
             } else {
-              file.url = data.Location
```

Comment:
Since `cloudflarePublicAccessUrl` is already required for CSP, I think it would be useful to strongly recommend having it set. I will update the README too

I can add an error and `process.emitWarning` (when `cloudflarePublicAccessUrl` isn't set) though.